File: sdk/go/examples/basic.go
package main

import (
	"fmt"
	"log"

	poubelle "github.com/poubelle/sdk-go"
)

func main() {
	client, err := poubelle.NewClient("poubelle://admin:admin@127.0.0.1:5432")
	if err != nil {
		log.Fatal(err)
	}

	fmt.Println("Connecting to Poubelle DB...")
	if err := client.Connect(); err != nil {
		log.Fatal(err)
	}
	defer client.Close()
	fmt.Println("Connected!")

	fmt.Println("\nCreating table...")
	result, err := client.Query("CREATE TABLE users (id INT, name TEXT, age INT)")
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(result)

	fmt.Println("\nInserting data...")
	client.Query("INSERT INTO users (id, name, age) VALUES (1, 'Alice', 30)")
	client.Query("INSERT INTO users (id, name, age) VALUES (2, 'Bob', 25)")
	client.Query("INSERT INTO users (id, name, age) VALUES (3, 'Charlie', 35)")
	fmt.Println("Inserted 3 rows")

	fmt.Println("\nQuerying data (debug format)...")
	rows, err := client.Execute("SELECT * FROM users")
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("Found %d rows:\n", len(rows))
	for _, row := range rows {
		fmt.Printf("  %v\n", row)
	}

	fmt.Println("\nQuerying data (JSON format)...")
	jsonRows, err := client.ExecuteJSON("SELECT * FROM users WHERE age > 25")
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("Found %d rows:\n", len(jsonRows))
	for _, row := range jsonRows {
		fmt.Printf("  %v\n", row)
	}

	fmt.Println("\nDone!")
}

File: crates/sdk/examples/basic.rs
use poubelle_sdk::PoubelleClient;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut client = PoubelleClient::new("poubelle://admin:admin@127.0.0.1:5432")?;

    println!("Connecting to Poubelle DB...");
    client.connect().await?;
    println!("Connected!");

    println!("\nCreating table...");
    let result = client
        .query("CREATE TABLE users (id INT, name TEXT, age INT)")
        .await?;
    println!("{}", result);

    println!("\nInserting data...");
    client
        .query("INSERT INTO users (id, name, age) VALUES (1, 'Alice', 30)")
        .await?;
    client
        .query("INSERT INTO users (id, name, age) VALUES (2, 'Bob', 25)")
        .await?;
    client
        .query("INSERT INTO users (id, name, age) VALUES (3, 'Charlie', 35)")
        .await?;
    println!("Inserted 3 rows");

    println!("\nQuerying data...");
    let rows = client.execute("SELECT * FROM users").await?;
    println!("Found {} rows:", rows.len());
    for row in rows {
        println!("  {:?}", row);
    }

    println!("\nClosing connection...");
    client.close().await?;
    println!("Done!");

    Ok(())
}
